reasoner/planner/OrderingCoster.java

```diff
@@ -79,6 +84,11 @@ LocalAllCallsCosting createAllCallsCosting(ReasonerPlanner.CallMode callMode, Li
                 double allAnswersForUnrestrictedMode = answerCountEstimator.localEstimate(conjunctionNode.conjunction(), resolvable, resolvableMode);
                 double cyclicScalingFactor = projectionVars.isEmpty() && allAnswersForUnrestrictedMode != 0 ? 0.0 :
                         (double) estimator.answerEstimate(projectionVars) / allAnswersForUnrestrictedMode;
+
+                // Terrible overestimate. Let's take the square-root to be more realistic.
+                if (allAnswersForUnrestrictedMode != 0) {
+                    cyclicScalingFactor = Math.min(cyclicScalingFactor, Math.sqrt(allAnswersForUnrestrictedMode) / allAnswersForUnrestrictedMode);
```

Comment:
Need to decide if this is still needed, given we've revised the answer count itself.